src/python/visualize_matching_result.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+import cv2
+import argparse
+from pathlib import Path
+import protos_io
+
+
+def createHiddenMatchImage(width, height):
+    image = np.zeros((height, width, 3), dtype=np.uint8)
+    text_left_corner = (int(width / 3), int(height / 2))
+    image = cv2.putText(
+        image,
+        "No match found",
+        text_left_corner,
+        cv2.FONT_HERSHEY_SIMPLEX,
+        1,
+        (255, 255, 255),
+        2,  # thickness
+    )
+    return image
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--matching_result",
+        required=True,
+        type=Path,
+        help="Path to the matching result .MatchingResult.pb file",
+    )
+    parser.add_argument(
+        "--query_images",
+        required=True,
+        type=Path,
+        help="Path to directory with query images.",
+    )
+    parser.add_argument(
+        "--reference_images",
+        required=True,
+        type=Path,
+        help="Path to directory with reference images.",
+    )
+    parser.add_argument(
+        "--output_dir",
+        required=True,
+        type=Path,
+        help="Output dir for the resulting images.",
+    )
+
+    args = parser.parse_args()
+
+    matching_result = protos_io.read_matching_result(args.matching_result)
+
+    args.output_dir.mkdir(exist_ok=True)
+
+    query_images = sorted(list(args.query_images.glob("*.jpg" or "*.png")))
+    reference_images = sorted(list(args.reference_images.glob("*.jpg" or "*.png")))
+
+    for match in matching_result.matches:
+        query_image = cv2.imread(query_images[match.query_id].as_posix())
```

Comment:
I'm not sure about the use of `as_posix()` here. It might be that using `str(query_images[match.query_id])` will suit your needs better if this ever has to run on a non-posix system. If you only aim at Linux or MacOS, then feel free to ignore this comment.

Reply:
Thanks, switched to `str`.